### PR TITLE
Fix for domain_user supporting port.

### PR DESCRIPTION
**Pull request by hoytak**
Allows domain:port:user to work in xet:// urls. Fix CLI-203

### CODE REVIEW

File: python/pyxet/pyxet/url_parsing.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def parse_url(url, default_domain=None, expect_branch = None, expect_repo = True
         ret.domain_explicit = False
         explicit_user = None
     else:
-        domain_user = netloc.split(":")
+        domain_user = netloc.rsplit(":", maxsplit = 2)
```
Comment (seanses) [inline python/pyxet/pyxet/url_parsing.py:184]:
```suggestion
        domain_user = netloc.rsplit(":", maxsplit = 1)
```

Comment (seanses) [reply python/pyxet/pyxet/url_parsing.py:184]: https://www.w3schools.com/python/ref_string_rsplit.asp#:~:text=%23%20setting%20the%20maxsplit%20parameter%20to%201%2C%20will%20return%20a%20list%20with%202%20elements!

File: python/pyxet/pyxet/url_parsing.py
Comment (seanses) [inline python/pyxet/pyxet/url_parsing.py:1]: Unfortunately this is not the full fix for `XET_ENDPOINT=http://localhost:3000` yet: `def remote()` and `def domain_url()` need to return `http://` urls.

File: python/pyxet/pyxet/url_parsing.py (hunk 1 above)
Comment (hoytak) [reply python/pyxet/pyxet/url_parsing.py:184]: Oh, I see. Of course.

File: python/pyxet/pyxet/url_parsing.py
Comment (hoytak) [reply python/pyxet/pyxet/url_parsing.py:1]: Isn't that also  http://endpoint:port/ ?

File: python/pyxet/pyxet/url_parsing.py (hunk 1 above)
Comment (hoytak) [reply python/pyxet/pyxet/url_parsing.py]: Fixed.

File: python/pyxet/pyxet/url_parsing.py
Comment (seanses) [reply python/pyxet/pyxet/url_parsing.py:1]: Those two functions assume `https://` scheme

Comment (hoytak) [reply python/pyxet/pyxet/url_parsing.py:1]: Ah, you mean that it needs to use http://, not https://.  Gotcha.  I see.

Comment (seanses) [reply python/pyxet/pyxet/url_parsing.py:1]: Low pri, we don't have to make drastic changes before the prod push tmr.

File: python/pyxet/scripts/run_tests.sh [hunk 2]
```diff
@@ -30,6 +30,6 @@ pip install target/wheels/pyxet-*.whl
 
 # TODO: This runs the tests in parallel using pytest-xdist
 # Error: tests in cli can't be run simultaneously actually, as there are conflicts.
-#pytest -n 12 --verbose tests/
```
Comment (seanses) [inline python/pyxet/scripts/run_tests.sh:33]: Is this desired given the above comment?

**Review by seanses — APPROVED** [review]
LGTM! Thanks for the fix

**Pull request merged**